src/socket.c

```diff
@@ -32,16 +32,15 @@
 void socket_init(socket_ctx *ctx, char *path) {
 	printf("initialize socket\n");
 	ctx->socket.sun_family = AF_UNIX;
-	strcpy(ctx->socket.sun_path, path);
+	strncpy(ctx->socket.sun_path, path, strlen(path));
 	unlink(path);
-	signal(SIGPIPE, SIG_IGN);
 	ctx->fd = socket(AF_UNIX, SOCK_STREAM | SOCK_NONBLOCK, 0);
 	bind(ctx->fd, (struct sockaddr *)&ctx->socket, sizeof(ctx->socket));
 	listen(ctx->fd, 5);
```

Comment:
The return values of `bind()` and `listen()` shouldn't be dropped silently. I would propose:

``` c
ret = bind(...);
if (ret == -1) {
    perror("bind");
    exit(EXIT_FAILURE);
}

ret = listen(...);
if (ret == -1) {
    perror("listen");
    exit(EXIT_FAILURE);
}
```